### ISSUE TITLE
tpu torch xla is not using xla_cache

### ISSUE THREAD

**Issue by Chenyaaang** [pytorch/xla#9094]
## 🐛 Bug

I run a same script twice, the first time it saves xla cache into a directory, the 2nd time it's not using the xla cache, instead, it generates new xla_cache, with same hash. The compilation time of the 2 runs are almost the same. 

## To Reproduce

```
import os
import torch
import time
import torch_xla.core.xla_model as xm
import torch_xla.runtime as xr

@torch.compile(backend="openxla", fullgraph=True, dynamic=False)
def select_hidden_states(hidden_states, indices_do_sample):
    return hidden_states[indices_do_sample]

def precompile_select_hidden_states(hsize = 4096, num_tokens_paddings=[16, 32, 64], num_reqs_paddings=[8, 16, 32]) -> None:
    device = xm.xla_device()
    start = time.perf_counter()
    for num_tokens in num_tokens_paddings:
        dummy_hidden = torch.zeros((num_tokens, hsize),
                                    device=device,
                                    dtype=torch.bfloat16)
        torch._dynamo.mark_dynamic(dummy_hidden, 0)
        for num_reqs in num_reqs_paddings:
            print(f"{num_tokens=}, {num_reqs=}")
            indices = torch.zeros(num_reqs,
                                    dtype=torch.int32,
                                    device=device)
            torch._dynamo.mark_dynamic(indices, 0)
            select_hidden_states(dummy_hidden, indices)
            # Requests can't be more than tokens.
            if num_reqs >= num_tokens:
                break
    xm.wait_device_ops()
    end = time.perf_counter()
    print("Compilation finished in %.2f [secs].", end - start)

xla_cache_path = os.path.expanduser("~/.cache/test_xla")
xr.initialize_cache(xla_cache_path, readonly=False)
precompile_select_hidden_states()
```
-->

1. Copy this script, replace `xla_cache_path` to a folder you have access read/write access. 
2. Run the script for the first time, you will see cache saved in the `xla_cache_path` folder, as below
The script took 1.25seconds, and xla_cache are saved on May 5, 19:59pm
<details>
(base) chenyangli_google_com@t1v-n-a23081af-w-0:/mnt/disks/persist$ ll ~/.cache/vllm/test_xla
total 1056
drwxrwxr-x 2 chenyangli_google_com chenyangli_google_com   4096 May  5 19:59 ./
drwxrwxr-x 5 chenyangli_google_com chenyangli_google_com   4096 May  5 19:59 ../
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 100055 May  5 19:59 142755492151925139001220011332256042622
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 133853 May  5 19:59 142947229373129781583056831763345279630
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  48164 May  5 19:59 179977713440745634561102714242062765079
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  40912 May  5 19:59 24393465306967763532772599988709587007
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  40912 May  5 19:59 253542937683851797340656852546505099671
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  98006 May  5 19:59 276315257657769070111974582679800920472
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 139485 May  5 19:59 284435209543584171258911109343191884653
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  81108 May  5 19:59 41954398619785526896952630472505815396
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  83158 May  5 19:59 59367263880333735213047576956600301586
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 105690 May  5 19:59 70152964789360659636750897487190971208
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  46114 May  5 19:59 81475962550748795711313333018260730022
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  88792 May  5 19:59 92494755209454192542651722019667944751
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  45085 May  5 19:59 96875099153576277337736008200492258639
</details>
3. Do not delete the xla cache or modify the script, run the script again. This time the script also took 1.25s, and the files in xla_cache_folder still have the exact same hash names, but are saved on May 5 20:00pm, which means these are cache miss, new cache are generated and saved. 
<details>
(base) chenyangli_google_com@t1v-n-a23081af-w-0:/mnt/disks/persist$ ll ~/.cache/vllm/test_xla
total 1056
drwxrwxr-x 2 chenyangli_google_com chenyangli_google_com   4096 May  5 20:00 ./
drwxrwxr-x 5 chenyangli_google_com chenyangli_google_com   4096 May  5 19:59 ../
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 100056 May  5 20:00 142755492151925139001220011332256042622
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 133854 May  5 20:00 142947229373129781583056831763345279630
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  48165 May  5 20:00 179977713440745634561102714242062765079
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  40913 May  5 20:00 24393465306967763532772599988709587007
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  40913 May  5 20:00 253542937683851797340656852546505099671
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  98007 May  5 20:00 276315257657769070111974582679800920472
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 139486 May  5 20:00 284435209543584171258911109343191884653
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  81109 May  5 20:00 41954398619785526896952630472505815396
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  83159 May  5 20:00 59367263880333735213047576956600301586
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com 105691 May  5 20:00 70152964789360659636750897487190971208
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  46115 May  5 20:00 81475962550748795711313333018260730022
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  88793 May  5 20:00 92494755209454192542651722019667944751
-rw-rw-r-- 1 chenyangli_google_com chenyangli_google_com  45086 May  5 20:00 96875099153576277337736008200492258639
</details>

## Expected behavior

The expected behavior is the xla cache in the 1st run being reused on the 2nd run. 

## Environment

 - Reproducible on XLA backend [CPU/TPU/CUDA]: TPU
 - torch version: torch==2.8.0.dev20250408
 - torch_xla version:
torch_xla[tpu, pallas] @ https://storage.googleapis.com/pytorch-xla-releases/wheels/tpuvm/torch_xla-2.8.0.dev20250408-cp311-cp311-linux_x86_64.whl ; python_version == "3.11"

**Comment by Chenyaaang** [comment]
@miladm Can you please take a look at this? VLLM needs this bug to be resolved, and it will reduce the startup time for a warmed up node by a lot. Right now a llama3-8b single chip model needs 120s to warmup, larger models needs more time.

**Comment by yaochengji** [comment]
More context, when trying to load the compiled cache, we got
```
W torch_xla/csrc/runtime/pjrt_computation_client.cc:680] Failed to deserialize executable: UNIMPLEMENTED: Deserializing serialized executable not supported.
```

**Comment by yaochengji** [comment]
@miladm is there any progress on this?

**Comment by Dogacel** [comment]
Issue still persists at torch 2.9.0
